### PR TITLE
Update readme to use mysql 8.0

### PR DESCRIPTION
**Pull request by tussosedan**
As our staging is now on mysql 8.0, it's a good time to update the readme so that all devs use it too!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,11 +29,11 @@ This repository contains all of the code that runs on [worldcubeassociation.org]
...
 ## Run directly with Ruby (lightweight, but only runs the Rails portions of the site)
 
-- Install either MySQL 5.6 or 5.7, and set it up with a user with username "root" with an empty password.
+- Install MySQL 8.0, and set it up with a user with username "root" with an empty password.
   If it poses problems, try the following:
   ```shell
-  # Install MySQL 5.6/5.7 if you haven't already.
-  # Using apt that would be: sudo apt install mysql-server-5.6
+  # Install MySQL 8.0 if you haven't already.
+  # Using apt that would be: sudo apt install mysql-server
```
Comment (jfly) [inline README.md]: Does `mysql-server-8.0` work here instead?

Comment (tussosedan) [reply README.md]: I thought of doing that at first, but it doesn't look like there's a package named like that?

https://packages.ubuntu.com/search?keywords=mysql-server&searchon=names&suite=disco&section=all

**Comment by viroulep** [comment]
Since we did already upgrade to MySQL 8, would you mind adjusting the readme to point to MySQL's documentation regarding how to install version 8.0 on various distributions?
The url is this one: https://dev.mysql.com/doc/refman/8.0/en/linux-installation.html
It has some very straightforward way using their custom package repository.

**Comment by tussosedan** [comment]
Good idea! Done.

**Pull request merged**